### PR TITLE
Standardization of the definition of env vars.

### PR DESCRIPTION
**Pull request by ldez**
Standardization of the definition of env vars:

- uses exported constants
- define standards for variable names

### CODE REVIEW

**Comment by wyattjoh** [comment]
This looks great! A suggestion however...

There's this super neat package, https://github.com/kelseyhightower/envconfig that allows you to use struct tags to define your configuration for reading from the environment where it'll also handle defaults. Might be interesting to use this rather than relying on our own library code in the future?

**Comment by dmke** [comment]
Package `envconfig` looks indeed nice. It also has zero external dependencies.

Could be worth investigating whether this is compatible with `./platform/tester/env.go`, and how much modification is needed. :thinking:

**Comment by ldez** [comment]
Some env vars are not really managed by lego but by the clients (azure, gcloud, route53, lightsail, ...)

And `envconfig` don't allows the users of lego as lib to get the env var names and reuse-it.

**Comment by ldez** [comment]
The modifications that I was able to make give the impression that it is homogeneous but it is not the case.

File: providers/dns/dynu/dynu.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func (d *DNSProvider) Present(domain, token, keyAuth string) error {
 		NodeName:   dns01.UnFqdn(strings.TrimSuffix(fqdn, dns.Fqdn(domain))),
 		TextData:   value,
 		State:      true,
-		TTL:        300,
+		TTL:        d.config.TTL,
```
[the remark this answers is not in the document]
Comment (dmke) [reply providers/dns/dynu/dynu.go]: (it may stay)

**Comment by dmke** [comment]
> Some env vars are not really managed by lego [...]

Good point.

> The modifications that I was able to make give the impression that it is homogeneous but it is not the case.

After reviewing all the cases, I can confirm. Especially the "weird" cases really would make the adoption of `envconfig` a mess.

File: providers/dns/autodns/autodns_test.go [hunk 2]
```diff
@@ -8,7 +8,13 @@ import (
...
 )
 
-var envTest = tester.NewEnvTest(envAPIEndpoint, envAPIUser, envAPIPassword)
+const envDomain = envNamespace + "DOMAIN"
+
+var envTest = tester.NewEnvTest(
+	EnvAPIEndpoint,
+	EnvAPIUser,
+	EnvAPIPassword).
+	WithDomain(envDomain)
```
[the remark this answers is not in the document]
Comment (ldez) [reply providers/dns/autodns/autodns_test.go]: the domain is used in `TestLivePresent` and `TestLiveCleanUp`

[the remark this answers is not in the document]
Comment (ldez) [reply providers/dns/autodns/autodns_test.go]: https://github.com/go-acme/lego/blob/aeb20f4debd7b16abd0478e5b139017dcf38ebf0/providers/dns/autodns/autodns_test.go#L132

https://github.com/go-acme/lego/blob/aeb20f4debd7b16abd0478e5b139017dcf38ebf0/providers/dns/autodns/autodns_test.go#L145

[the remark this answers is not in the document]
Comment (dmke) [reply providers/dns/autodns/autodns_test.go]: Oh, okay. Never mind then :-)

**Comment by ldez** [comment]
@dmke thank you to review this huge and "robotic" PR :smile:

**Pull request merged**